crates/fluvio-cli/src/client/consume/mod.rs

```diff
@@ -115,17 +115,21 @@ mod cmd {
         #[clap(long, conflicts_with_all = &["key-value", "format"])]
         pub table_format: Option<String>,
 
-        /// Consume records starting X from the beginning of the log (default: 0)
-        #[clap(short = 'B', value_name = "integer", conflicts_with_all = &["offset", "tail"])]
-        pub from_beginning: Option<Option<u32>>,
-
         /// The offset of the first record to begin consuming from
         #[clap(short, long, value_name = "integer", conflicts_with_all = &["from-beginning", "tail"])]
         pub offset: Option<u32>,
 
-        /// Consume records starting X from the end of the log (default: 10)
-        #[clap(short = 'T', long, value_name = "integer", conflicts_with_all = &["from-beginning", "offset"])]
-        pub tail: Option<Option<u32>>,
+        /// Consume records starting X from the beginning of the log (default:0, change X with -n)
+        #[clap(short = 'B', long="head", group = "can_be_offset", conflicts_with_all = &["offset", "tail"])]
+        pub from_beginning: bool,
+
+        /// Consume records starting X from the end of the log (default: 10, change X with -n)
+        #[clap(short = 'T', long,  group = "can_be_offset", conflicts_with_all = &["from-beginning", "offset"])]
+        pub tail: bool,
+
+        /// --head consume beginning X after start of log. --tail consume beginning X before end of log.
+        #[clap(short = 'n', value_name = "integer", requires("can_be_offset"))]
+        pub amount_to_offset: Option<u32>,
```

Comment:
These sound like good changes to me. 

### Flags
One concern that has been nagging at me is the mixing of **relative** and **absolute** offsets. The current `-o/--offset` option mapped to `offset` in the struct is an absolute offset, presumably from the beginning of the topic regardless of whether the log has ever been truncated. I made a few additional changes to try and make the distinction more clear, lmk what you think.

1. `--offset 0` -> `--start`
2. `--end-offset` -> `--end`
3. `-B/--head` -> `-H/--head`
4. `--offset 5` -> `--start -n 5`

Also, the `end-offset` option conflicts with `tail` but I don't see why it should need to... say the current offset is 100 and you want to consume 99-150, passing `--tail -n 1 --end-offset 150` would be a valid setup.

### Behavior
Also, seems like the cli should abort here instead of continuing to process.
```
# flvd consume  --start -n 1 --end 0 echo 
Argument end-offset must be greater than or equal to specified offset
Consuming records from offset 1 in topic 'echo'
hello1
⠁
End-offset has been reached; exiting
Consumer stream has closed
```

Possible current errors are 
```
    IoError(IoError),
    EnvError(env::VarError),
    JsonError(serde_json::Error),
    DiffError(DiffError),
    HttpError(HttpError),
    InvalidHttpHeader(InvalidHeaderValue),
    K8ConfigError(ConfigError),
    PatchError,
    HyperError(HyperError),
    Client(StatusCode),
    Other(String),
    #[cfg(feature = "openssl_tls")]
    Tls(fluvio_future::openssl::TlsError),
```
Would `Client(1)` be a reasonable choice here?


One more todo here:
```
# flvd consume --start --end 0 echo
Consuming records from offset 0 in topic 'echo'
hello
⠁
End-offset has been reached; exiting
Consumer stream has closed
```
`Consuming records from offset 0 in topic 'echo'` should probably read `from offset 0 to offset 0...`

Reply:
Flag changes look great, very clear.

As for the error, good find. The your first example when the end offset < the start offset, seems like we should be able to exit before even opening the stream. 

> passing --tail -n 1 --end-offset 150 would be a valid setup.

With `tail` and `end-offset`, combining them would be nice. I think the only thing stopping that use is not having any check for the end < start offset condition. Seems that it was probably low priority to handle.

> Would Client(1) be a reasonable choice here?

I usually opt for `Other(...)` just so I can give an error message. I'm not sure there are any strong conventions used in the CLI.

> from offset 0 to offset 0...
This would be my expectation as well

Also, sorry for the delayed response! I didn't see your new changes, and the last week the team was focused on supporting our people at KubeCon.